core/src/main/java/io/questdb/griffin/SqlOptimiser.java

```diff
@@ -3843,13 +3843,16 @@ private QueryModel rewriteSelectClause0(final QueryModel model, SqlExecutionCont
         }
 
         boolean outerVirtualIsSelectChoose = true;
-        //if there are explicit group by columns then nothing else should go to group by model
-        //select columns should either match group by columns exactly or go to outer virtual model
+        // if there are explicit group by columns then nothing else should go to group by model
+        // select columns should either match group by columns exactly or go to outer virtual model
         ObjList<ExpressionNode> groupBy = groupByModel.getGroupBy();
         boolean explicitGroupBy = groupBy.size() > 0;
 
         if (explicitGroupBy) {
-            // Outer model is not needed only if select clauses is the same as group by plus aggregate function calls   
+            groupByAliases.clear();
```

Comment:
The bugfix is here. All other changes in this file are code clean up related.

Reply:
I think it'd be a bit better to put both lines above conditional to avoid potential problems with implicit group by (both groupByAliases and groupByNodes are used in select list processing - line 3956 ).

Reply:
Done in dd83f9104fe7b7474af9d7ba1d3fe13e8dd51547